src/converter/one/jfr/JfrReader.java

```diff
@@ -189,12 +202,14 @@ public <E extends Event> E readEvent(Class<E> cls) throws IOException {
         return null;
     }
 
-    private ExecutionSample readExecutionSample() {
+    private ExecutionSample readExecutionSample(boolean hasSamples) {
         long time = getVarlong();
         int tid = getVarint();
         int stackTraceId = getVarint();
         int threadState = getVarint();
-        return new ExecutionSample(time, tid, stackTraceId, threadState);
+        getVarint(); // contextId
```

Comment:
This is a manual addition and JFR conversion doesn't work without this. AFAIK in the current usage of the library this only affects tests, Pyroscope uses https://github.com/grafana/jfr-parser to ingest JFR files which does the same.